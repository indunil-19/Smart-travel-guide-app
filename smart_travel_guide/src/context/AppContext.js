import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}