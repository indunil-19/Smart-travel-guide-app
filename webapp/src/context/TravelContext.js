import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 