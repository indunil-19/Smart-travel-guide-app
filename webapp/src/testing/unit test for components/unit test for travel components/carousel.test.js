import React from "react"
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import { TravelProvider } from "../../../context/TravelContext";
import "@testing-library/jest-dom/extend-expect";
import 'whatwg-fetch'
import Carousal from "../../../components/TravelPlanApp/carousal";
import { getByTestId, waitFor } from "@testing-library/dom";

describe('carousel test', () => {
    
    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><Carousal/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><Carousal photos={["123"]}  /></TravelProvider></MemoryRouter>,div)
        waitFor(()=>{
            expect(getByTestId("image")).toBeInTheDocument()
        })
       
    })
})
