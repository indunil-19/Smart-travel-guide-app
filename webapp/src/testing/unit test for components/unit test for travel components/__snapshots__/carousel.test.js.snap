// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provinces page test should match snapshot 1`] = `null`;
